x-pack/legacy/plugins/actions/server/actions_config.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { fromNullable } from 'fp-ts/lib/Option';
+import { URL } from 'url';
+import { asOk, asErr, Result } from './builtin_action_types/lib/result_type';
+
+export enum WhitelistedHosts {
+  Any = '*',
+}
+
+export interface ActionsKibanaConfig {
+  enabled: boolean;
+  whitelistedHosts: WhitelistedHosts.Any | string[];
```

Comment:
It seems slightly strange to have the type be either the string '*' or an array of strings; like it should always just be an array of strings, or maybe a single string or array of strings, but I generally am not a fan of variant param types.  For example, I could imagine someone moving from the default of `[]` to whitelisting everything, to try `["*"]`; or maybe if they've set the whitelist already to `*`, and deciding they only want to whitelist one hostname, changing that to `api.pagerduty.com` or whatever.  The latter would not fit the type above, the former might work but not clear to me (yet, because still reading :-) ).

Reply:
The reason I didn't put the ANY in the array was because it keeps the code simpler (otherwise you compare to ANY in the loop and it's less explicit.
But I think this is subjective... I prefer variant param types when you have a typesystem to check you're doing something valid :)

In the latter case, btw, it wouldn't pass validation so they'd get an error and hopefully the documentation would make it clear.

Reply:
Are there existing config values that are multi-typed in this fashion?  To me, there's the customer's cognitive overload to consider - now THEY have to know about the multi-valued type; seems easier if they just have a single type to consider.

I think if we keep this structure, we'd **still** want to check `*` in the array items, because it seems natural they may try to go from the default of `[]`, which is presumably the default :-), to `["*"]` without knowing all the rules, and we'd want that to be the same as `*`.  I could see `["*", "pagerduty.com"]` issuing an warn-level message about the spurious `pageduty.com` entry not being needed, the `*` entry whitelists everything.

Reply:
It's interesting... To me that's more confusing. 🤷‍♂️

But I'm less familiar with what is the norm in Kibana... I'll change it.

Reply:
I'm not familiar with all the Kibana (and elasticsearch, really) options, maybe there _is_ a pre-existing config that works similar to this?  Dunno.

Reply:
I can't see other APIs that currently support this.

I've changed it, though, to me this is a more confusing API and makes it easier to express problematic configurations that could mean all hostnames are whitelisted without the admin realising it, because it is far less explicit.

I'm a big believer in using the typesystem to make it impossible to express problematic (or invalid) states but I don't see how we can do this without committing fully to Typescript, so for now I'll stick to using this only on internals. So, in this case, I'll go with the flow.